maxq10xx/wolfssl-lwip-client.c

```diff
@@ -0,0 +1,318 @@
+        }
+    }
+#endif
+    /* Validate peer certificate */
+    wolfSSL_CTX_set_verify(ctx, WOLFSSL_VERIFY_PEER, NULL);
+
+    /* Create a WOLFSSL object */
+    if ((*ssl = wolfSSL_new(ctx)) == NULL) {
+        MXC_ERR_MSG("ERROR: failed to create WOLFSSL object\n");
+        ret = -1;
+        exit_clean = 1;
+        goto exit;
+    }
+exit:
+    /* Cleanup and return */
+    if (ctx != NULL)
+        wolfSSL_CTX_free(ctx);
+    if (exit_clean)
+        mxc_wolfssl_close(sockfd, *ssl);
+}
+
+void mxc_wolfssl_connect(int sockfd, WOLFSSL *ssl)
+{
+    int ret, err;
+    int exit_clean = 0;
+
+#ifdef WOLFSSL_TLS13
+    if (alg == RSA) {
+        if (wolfSSL_UseKeyShare(ssl, WOLFSSL_FFDHE_2048) != WOLFSSL_SUCCESS) {
+            MXC_ERR_MSG("ERROR: failed to create WOLFSSL object\n");
+            ret = -1;
+            exit_clean = 1;
+            goto exit;
+        }
+    }
+#endif /* WOLFSSL_TLS13 */
+    //MXC_DEBUG_MSG("mxc_wolfssl_connect->wolfSSL_set_fd\n");
+    /* Attach wolfSSL to the socket */
+    if ((ret = wolfSSL_set_fd(ssl, sockfd)) != WOLFSSL_SUCCESS) {
+        MXC_ERR_MSG("ERROR: Failed to set the file descriptor\n");
+        exit_clean = 1;
+        goto exit;
+    }
+
+    wolfSSL_SSLSetIORecv(ssl, mxc_wolfssl_receive);
+    wolfSSL_SSLSetIOSend(ssl, mxc_wolfssl_send);
+
+    //MXC_DEBUG_MSG("mxc_wolfssl_connect->wolfSSL_connect\n");
+    /* Connect to wolfSSL on the server side */
+    do {
+        ret = wolfSSL_connect(ssl);
+        err = wolfSSL_get_error(ssl, ret);
+    } while (err == WC_PENDING_E);
+    if (ret != WOLFSSL_SUCCESS) {
+        MXC_ERR_MSG("ERROR: failed to connect to wolfSSL\n");
+        exit_clean = 1;
+        goto exit;
+    }
+
+    MXC_DEBUG_MSG_GRN("\nmxc_wolfssl_connect-> Done\n\n");
+
+exit:
+    if (exit_clean)
+        mxc_wolfssl_close(sockfd, ssl);
+}
+
+void mxc_wolfssl_write(WOLFSSL* ssl, const void* data, int sz)
+{
+    int ret;
+
+    MXC_DEBUG_MSG(". mxc_wolfssl_write...\n");
+
+    /* Send the message to the server */
+    if ((ret = wolfSSL_write(ssl, data, sz)) != sz) {
+        MXC_ERR_MSG("ERROR: failed to write entire message\n");
+        MXC_ERR_MSG("%d bytes of %d bytes were sent", ret, sz);
+    }
+}
+
+int mxc_wolfssl_read(WOLFSSL* ssl)
+{
+    int ret;
+    char buff[SOCKET_BUF_SIZE];
+
+    /* Read the server data into our buff array */
+    memset(buff, 0, sizeof(buff));
+    if ((ret = wolfSSL_read(ssl, buff, sizeof(buff)-1)) == -1) {
+        MXC_ERR_MSG("ERROR: failed to read\n");
+    }
+
+    /* Print to stdout any data the server sends */
+    MXC_DEBUG_MSG("Server: %s\n", buff);
+    return ret;
+    // ret = 0; /* success */
+}
+
+void mxc_wolfssl_close(int sockfd, WOLFSSL* ssl)
+{
+    unsigned char lBuff[SOCKET_BUF_SIZE];
+    /* Cleanup and return */
+    if (ssl != NULL)
+        wolfSSL_free(ssl);
+
+    wolfSSL_Cleanup();
+
+    if (sockfd >= 0)
+        lwip_close(sockfd);
+    while(lwip_read(sockfd,lBuff, SOCKET_BUF_SIZE) > 0);

```

Comment:
Why is this after the close of the socket? Is it to flush it after the socket has already been closed? Nit pick: there is no space after first `,`

Reply:
Waiting for feedback from someone else.

Reply:
According the feedback received, this is correct. The socket is being flushed.